lib/keymap-helpers.nix

```diff
@@ -107,9 +107,10 @@ rec {
           (lib.optionalAttrs (isAttrs key || key) {
             key = lib.mkOption (
               {
-                type = types.str;
+                type = with types; nullOr str;
                 description = "The key to map.";
                 example = "<C-m>";
+                default = null;
```

Comment:
If we do this, then this could be `defaults.key or null` instead of the optionalAttrs below.

I'm wondering if it makes more sense to check the submodule's `_module.args.name` and use that as the default?

That way when the submodule is used in an `attrsOf` type, the attr name would be the default for the key option.